### ISSUE TITLE
Questions about the dataset at train.py

### ISSUE THREAD

**Issue by JinSJtu** [lyxok1/STM-Training#13]
For loss :cycle-loss += criterion(first_out, mask[0:1], num_objects, ref = mask[t:t+1,:num_objects]).
Does ref mean that the entire training set needs to be labeled? Thanks！

**Comment by lyxok1** [comment]
Yes, since the model is trained on sampled frames, we should each frame is annotated with at least one object, ideally, all sampled frames should contain at least one common object

**Comment by JinSJtu** [comment]
Thanks for your answer! I have run the model on my own dataset successfully. But i found a very strange phenomenon during testing on long video. If the model uses only the first frame as memory, the segmentation result of the model has almost the same segmentation accuracy as the way memory is added every five frames (Dice>0.9). Have you investigated the effect of different memory frame counts on segmentation accuracy?
